packages/core/src/routes/authn.ts

```diff
@@ -144,4 +150,73 @@ export default function authnRoutes<T extends AnonymousRouter>(
       return next();
     }
   );
+
+  /**
+   * SAML SSO connector's assertion consumer service endpoint
+   *
+   * @param connectorId The connector id.
+   * @property body The SAML assertion response body.
+   * @property body.RelayState We use this to find the connector session.
+   * @returns Redirect to the redirect uri find in the connector session storage.
+   *
+   * @remark
+   * This API is used to handle SSO SAML assertion from the identity provider.
+   * Validate and parse the SAML assertion, then store the assertion data to the connector session storage.
+   * Redirect to the redirect uri find in the connector session storage.
+   * The above endpoint will be deprecated in the future.
+   */
+  router.post(
+    `/authn/${ssoPath}/saml/:connectorId`,
+    koaGuard({
+      body: z.object({ RelayState: z.string() }).catchall(z.unknown()),
+      params: z.object({ connectorId: z.string().min(1) }),
+      status: [302, 404, 500],
+    }),
+    async (ctx, next) => {
+      const {
+        params: { connectorId },
+        body,
+      } = ctx.guard;
+
+      // Will throw 404 if connector not found, or not supported
+      const connectorData = await ssoConnectorsLibrary.getSsoConnectorById(connectorId);
+      const { providerName } = connectorData;
+
+      // Get relay state from the request body. We need to use it to find the connector session.
+      // All the rest of the request body will be validated and parsed by the connector.
+      const { RelayState: jti } = body;
+
+      // Retrieve the single sign on session data using the jti
+      const singleSignOnSession = await getSingleSignOnSessionResultByJti(jti, provider);
+
+      const { redirectUri, state, connectorId: sessionConnectorId } = singleSignOnSession;
```

Comment:
nit: can add comments to show why we need to check `sessionConnectorId` and attach `state` when redirecting. Seems this can not be directly understood by reviewers without too much context.

Reply:
`state` comment added. `sessionConnectorId` assertion logic with error code looks straight forward.